### PR TITLE
seeds: add violet subdomain dummy data

### LINKED ISSUE #628: create better seeds
we should seed the development environment to look like `violet.restarone.solutions`

make it look lived in: 

1. many user accounts 
2. forum threads/posts
3. emails / threads
4. API namespaces / forms 
5. Blog posts
6. Subdomain setup with logos and favicons etc

this needs to be automated to work with `rails db:seed`

### PR DESCRIPTION
**Pull request by yashkir**
resolves #628 

- seed to look like violet.restarone.solutions
- add faker data

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -6,4 +9,131 @@
...
+# seeds.rb gets run for each tenant, so we must make sure to run
+# Subdomain.create only once here.
+if Apartment::Tenant.current == 'public'
+  Subdomain.unsafe_bootstrap_www_subdomain
+  Subdomain.create!(name: 'violet')
+end
+
+if Apartment::Tenant.current == 'violet'
+  user_orange = User.create!(
+    email: 'orange@rails.com',
```
Comment (donrestarone) [inline db/seeds.rb:22]: please use `Faker::Internet.email `

File: db/seeds.rb [hunk 2]
```diff
@@ -6,4 +9,131 @@
...
+# Subdomain.create only once here.
+if Apartment::Tenant.current == 'public'
+  Subdomain.unsafe_bootstrap_www_subdomain
+  Subdomain.create!(name: 'violet')
+end
+
+if Apartment::Tenant.current == 'violet'
+  user_orange = User.create!(
+    email: 'orange@rails.com', 
+    name: 'Orange Tester',
```
Comment (donrestarone) [inline db/seeds.rb:23]: please use `Faker::Name.name`

File: db/seeds.rb [hunk 3]
```diff
@@ -6,4 +9,131 @@
...
+    name: 'Red Tester',
+    password: '123456', 
+    password_confirmation: '123456', 
+    global_admin: false, 
+    confirmed_at: Time.now
+  )
+  user_red.update!(User::FULL_PERMISSIONS)
+
+  user_blue = User.create!(
+    email: 'blue@rails.com',
```
Comment (donrestarone) [inline db/seeds.rb:42]: same as same as https://github.com/restarone/violet_rails/pull/874/files#r913738056

File: db/seeds.rb [hunk 4]
```diff
@@ -6,4 +9,131 @@
...
+    password: '123456', 
+    password_confirmation: '123456', 
+    global_admin: false, 
+    confirmed_at: Time.now
+  )
+  user_red.update!(User::FULL_PERMISSIONS)
+
+  user_blue = User.create!(
+    email: 'blue@rails.com', 
+    name: 'Blue Tester',
```
Comment (donrestarone) [inline db/seeds.rb:43]: same as https://github.com/restarone/violet_rails/pull/874/files#r913739531

File: db/seeds.rb [hunk 5]
```diff
@@ -6,4 +9,131 @@
...
+  blog_post = site.blog_posts.create!(
+    title: "My first post",
+    layout: site.layouts.find_by(identifier: 'default')
+  )
+  Comfy::Cms::Fragment.create!(
+    identifier: 'content',
+    record: blog_post,
+    tag: 'wysiwyg',
+    content: "
+      <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quae duo sunt, unum facit. Uterque enim summo bono fruitur,
```
Comment (donrestarone) [inline db/seeds.rb:95]: please use `Faker::GreekPhilosophers.quote ` between paragraph/markup tags

File: db/seeds.rb [hunk 6]
```diff
@@ -6,4 +9,131 @@
...
+
+      <p>Laboro autem non sine causa; Memini vero, inquam; Bonum incolumis acies: misera caecitas. </p>
+
+      <p>In schola desinis. De quibus cupio scire quid sentias. </p>
+    "
+  )
+
+  # Forum
+  # -----
+  ForumCategory.create!(name: "cats", slug: "cats2", color: "#0000CC")
```
Comment (donrestarone) [inline db/seeds.rb:107]: please use `Faker::Educator.subject`

File: db/seeds.rb [hunk 7]
```diff
@@ -6,4 +9,131 @@
...
+  ForumCategory.create!(name: "dogs", slug: "dogs2", color: "#00CCCC")
+
+  thread = user.forum_threads.create!(title: "Cats are great?", forum_category: ForumCategory.first)
```
Comment (donrestarone) [inline db/seeds.rb:110]: please use `Faker::Educator.course_name`

File: db/seeds.rb [hunk 8]
```diff
@@ -6,4 +9,131 @@
...
+  post = user.forum_posts.create!(forum_thread: thread, body: "I Think they are awesome!")
+  post = user_orange.forum_posts.create!(forum_thread: thread, body: "I Think they are not so good!")
+  # TODO attachments
+
+  # Emails
+  # ------
+  email_thread = MessageThread.create!(recipients: ['violet@rails.com', 'red@rails.com'] )
```
Comment (donrestarone) [inline db/seeds.rb:117]: please use `Faker::Internet.email `

File: db/seeds.rb [hunk 9]
```diff
@@ -6,4 +9,131 @@
...
+  email_thread.messages.create(content: "An invitation to a meeting. Please come here at 4pm on Friday.")
```
Comment (donrestarone) [inline db/seeds.rb:118]: @yashkir this is a rich text field. you should use some markup and use `Faker::Movie.quote` to populate it

File: db/seeds.rb [hunk 10]
```diff
@@ -6,4 +9,131 @@
...
+  email_thread.messages.create(content: "I will be there", from: 'violet@rails.com')
+  email_thread.messages.create(content: "Have to skip this one.", from: 'red@rails.com')
+
+  email_thread = MessageThread.create!(recipients: ['violet@rails.com'] )
+  email_thread.messages.create(content: "I really like your site!", from: 'violet@rails.com')
```
Comment (donrestarone) [inline db/seeds.rb:123]: please use `Faker::Movie.quote `

**Comment by donrestarone** [comment]
Great first effort @yashkir -- lets use Faker along with the other changes I suggested above. After you attach screenshots of the Freshly Seeded State :seedling: we can deploy this to `staging`

**Comment by yashkir** [comment]
Added Faker and removed the subdomain. Should be fairly straightforward to extend these with fancier markup/date if needed:

Screenshot_2022-07-09_22-59-48
Screenshot_2022-07-09_23-00-03
Screenshot_2022-07-09_22-58-14
Screenshot_2022-07-09_23-13-00
Screenshot_2022-07-09_22-57-45
Screenshot_2022-07-09_23-11-39
Screenshot_2022-07-09_23-00-29

File: db/violet_seeds/violet.rb [hunk 11]
```diff
@@ -0,0 +1,1201 @@
...
+					</div>
+					<h1 class="text-capitalize">A Feature Rich <br><span class="text-accent">Web Platform</span><br>
+					</h1>
+					<p class="lead mt-3"> For hosting your next big idea
+					</p>
+				</div>
+				<div class="d-flex align-items-center my-5">
+					<a href="/forum" class="btn btn-lg btn-primary btn-shadow">Forum</a>
+					<a href="https://github.com/restarone/violet_rails" class="d-flex align-items-center btn btn-lg font-weight-bold bg-transparent">
+					<img src="{{ cms:file_link 31 }}" class="img mr-2">
```
Comment (donrestarone) [inline db/violet_seeds/violet.rb:612]: @yashkir where we use tokens like `{{ cms:file_link 31 }}` how are we ensuring these hard coded ID's (eg: `31`) match what is in the file system?

**Comment by donrestarone** [comment]
@yashkir Faker looks good! Can you attach some pictures of the web pages? Eg: `violet.restarone.solutions` landing page

**Comment by yashkir** [comment]
> @yashkir Faker looks good! Can you attach some pictures of the web pages? Eg: `violet.restarone.solutions` landing page

As mentioned in the review reply above, the images are not imported and show as blank. We could leave this as is, or import a minimal set (e.g. just the violet rails/restarone logos).

Screenshot_2022-07-10_23-11-40
Screenshot_2022-07-10_23-12-08
Screenshot_2022-07-10_23-12-23

**Comment by donrestarone** [comment]
@yashkir good call on it bloating source. But I think we should include it and strip it later (as part of a different task) during deployment. 

Can you go ahead and include the assets as well? So it looks just like `violet.restarone.solutions`

**Comment by donrestarone** [comment]
@yashkir Since CI is passing and it looks like you have linked assets, I'm creating a base branch for you. Please use that branch for changes (if any) moving forward 👍

**Pull request merged**